### ISSUE TITLE
Update channels have changed

### ISSUE THREAD

**Issue by LiamDawe** [ValveSoftware/SteamOS#1142]
With the latest SteamOS 3.5, it seems the way you set the update channels has changed (again?).

Originally the Steam Deck had two drop-down boxes for updates, then you simplified it to one that just had Stable / Beta / Preview:

![download (1)](https://github.com/ValveSoftware/SteamOS/assets/83223/04c4177c-fbd4-4824-96e3-6eb67bc38796)

Now mine seems stuck again with OS Update Channel and Steam Client Update Channel separately.

![download](https://github.com/ValveSoftware/SteamOS/assets/83223/2fe34bc0-b79a-4b21-bcc7-616232fce19f)

Is this now intentional again? As it wasn't actually noted anywhere I can see, I assume it's a bug with the update channels?

**Comment by TTimo** [comment]
This is the UI when enabling `Show Advanced Update Channels` under the `Developer` settings. Is this setting off for you?

**Comment by LiamDawe** [comment]
Yes, that setting is turned _off_. I did note that above, sorry if I wasn't clear enough on that!

![download (3)](https://github.com/ValveSoftware/SteamOS/assets/83223/99abb908-46d7-4179-b898-4a7ada6e8288)

**Comment by lostgoat** [comment]
That can happen if you are opted into a combination of settings that doesn't match any simplified "channel".

If you opt into stable client + stable OS / beta OS + beta client it should go back to a single dropdown.

Please note that client stable + OS beta is not a supported/tested combination and it may have unexpected bugs. When running SteamOS beta one should always be using the beta steam client. Which is the equivalent of the preview channel.

**Comment by LiamDawe** [comment]
That doesn't make sense though @lostgoat, since the standard update system is a single drop-down box of Stable, Beta and Preview and any time there's a Preview update the messaging is always to tell people to just go into Preview. Like the official announcement stating that (like all other updates since the move to 1 dropdown box).

Anyway, just a clip of it jumping from 1 to 2 boxes if it helps:

https://github.com/ValveSoftware/SteamOS/assets/83223/a87af4da-1c4c-4767-a576-36ccc3dcc9fe

**Comment by lostgoat** [comment]
Sorry, can you let me know which part doesn't make sense and I can probably clarify. There is probably some kind of misunderstanding that we can help clear up.

I *think* that the root cause of the misunderstanding is that there is a notion that there is an update channel called "Preview". However, "Preview" is just a concept in the UI it doesn't actually exist in the update servers. When you select any of the following options in the unified dropdown this is what you get:
  * Stable: Steam Client Stable + OS Stable
  * Beta: Steam Client Beta + OS Stable
  * Preview: Steam Client Beta + OS Beta

Let me know if that makes sense.

Also please note that "Show Advanced Update Channels" is a setting intended to be used by SteamOS developers and not end users. The separate options lead to very confusing messaging and also some unsupported combinations, which is why we unified them.  However, I do agree we need some improvements to how the UI works when the advanced options are enabled, as a significant amount of users opted into SteamOS main to get the color adjustment/SMT fixes early.

And just to re-iterate. Once you switch your steam client update channel to beta you should see the unified dropdown show up.

**Comment by LiamDawe** [comment]
The confusion here is simple: I went from Stable to Preview, the normal user way of updating, and now I have the two boxes when it's supposed to be one.

Sounds like I haven't explained it well enough in the OP. I'm also well aware advanced update channels is supposed to be for developers, as I've stated twice already in the OP and in a comment to TTimo mine is off :+1:

**Comment by lostgoat** [comment]
Has it ever been on? As mentioned above setting them to off after it has been on can lead to issues.

If the answer to the above is no, then it is also possible that something opted you out of the steam client beta after you were in preview. If you go to Settings->System->Create System Report and submit that to steam support I can check for clues as to what went wrong.

If you have any recollection of anything that may have edited the steam client beta opt in (using the desktop client or editing files under ~/.steam or ~/.local/share/Steam) let me know. As that could also be what caused it.

Overall it sounds like we need to improve the sate of things for users that end up in this state to make it easier to opt out. So thanks for bringing this up as an issue.

**Comment by LiamDawe** [comment]
Thanks for the reply. 

Yeah, it had been on Main before, but everything was reverted back to Stable well _before_ doing the Stable to Preview update. So I guess something, somewhere, got stuck?

**Comment by lostgoat** [comment]
We'll probably need more data to see what happened. Once you submit the system report I can take a look at what happened.

After you submit your system report let me know your steam account name or a link to your profile. If you'd prefer to send that privately lmk.

**Comment by LiamDawe** [comment]
Hopefully this is right: 
[systemreport-20230918210451.txt](https://github.com/ValveSoftware/SteamOS/files/12653499/systemreport-20230918210451.txt)

Profile:

**Comment by lostgoat** [comment]
Thanks for the info. We'll be changing how we handle the branch selection UI to better handle this case.

**Comment by LiamDawe** [comment]
Great to hear. Can help test any time :)

**Comment by Spidy123222** [comment]
> Thanks for the info. We'll be changing how we handle the branch selection UI to better handle this case.

I feel I should mention that the naming when having advanced update channels kinda gets confusing as stable users get stable beta and preview but when using the feature enabled in developer mode preview changes to beta channel. The naming should be more consistent. Maybe something like this could be done.

Stable, beta, preview, preview candidate, main (or renamed to nightly or whatever)
